internal/provider/kubernetes/kubernetes_test.go

```diff
@@ -154,15 +159,8 @@ func testGatewayClassAcceptedStatus(ctx context.Context, t *testing.T, provider
 func testGatewayClassWithParamRef(ctx context.Context, t *testing.T, provider *Provider, resources *message.ProviderResources) {
 	cli := provider.manager.GetClient()
 
-	// Create the namespace for the test case.
 	// Note: The namespace for the EnvoyProxy must match EG's configured namespace.
 	testNs := config.DefaultNamespace
-	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNs}}
-	require.NoError(t, cli.Create(ctx, ns))
```

Comment:
so we moved the namespace creation from one test to the other ?

Reply:
actually I think both tests needs the namespace, let me check it

Reply:
Since all tests now require a namespace for leader election(k8s uses that), we initialize it within TestProvider, which subsequently initiates the other tests. This is why the namespace init was removed from testGatewayClassWithParamRef, it can just reuse the namespace.

```go
	testcases := map[string]func(context.Context, *testing.T, *Provider, *message.ProviderResources){
		"gatewayclass controller name":         testGatewayClassController,
		"gatewayclass accepted status":         testGatewayClassAcceptedStatus,
		"gatewayclass with parameters ref":     testGatewayClassWithParamRef,
		"gateway scheduled status":             testGatewayScheduledStatus,
		"httproute":                            testHTTPRoute,
		"tlsroute":                             testTLSRoute,
		"stale service cleanup route deletion": testServiceCleanupForMultipleRoutes,
	}
```

Reply:
sounds good thanks